src/RobotInterface/include/BipedalLocomotion/RobotInterface/ISensorBridge.h

```diff
@@ -0,0 +1,318 @@
+     * @return true/false in case of success/failure
+     */
+    virtual bool getJointPosition(const std::string& jointName,
+                                  double& jointPosition,
+                                  double* receiveTimeInSeconds = nullptr) = 0;
+
+    /**
+     * Get all joints' positions
+     * @param[out] parameter all joints' position in radians
+     * @param[out] receiveTimeInSeconds time at which the measurement was received
+     * @return true/false in case of success/failure
+     */
+    virtual bool getJointPositions(GenericContainer::Vector<double>& jointPositions,
+                                   double* receiveTimeInSeconds = nullptr) = 0;
+
+    /**
+     * Get joint velocity
+     * @param[in] jointName name of the joint
+     * @param[out] jointVelocity joint velocity in radians per second
+     * @param[out] receiveTimeInSeconds time at which the measurement was received
+     * @return true/false in case of success/failure
+     */
+    virtual bool getJointVelocity(const std::string& jointName,
+                                  double& jointVelocity,
+                                  double* receiveTimeInSeconds = nullptr) = 0;
+
+    /**
+     * Get all joints' positions
+     * @param[out] parameter all joints' velocities in radians per second
+     * @param[out] receiveTimeInSeconds time at which the measurement was received
+     * @return true/false in case of success/failure
+     */
+    virtual bool getJointVelocities(GenericContainer::Vector<double>& jointVelocties,
+                                    double* receiveTimeInSeconds = nullptr) = 0;
+
+    /**
+     * Get IMU measurement
+     * The serialization of measurments is as follows,
+     *   (rpy acc omega mag)
+     *  - rpy in radians Roll-Pitch-Yaw Euler angles
+     *  - acc in m/s^2 linear accelerometer measurements
+     *  - omega in rad/s gyroscope measurements
+     *  - mag in tesla magnetometer measurements
+     * @param[in] imuName name of the IMU
+     * @param[out] imuMeasurement imu measurement of size 12
+     * @param[out] receiveTimeInSeconds time at which the measurement was received
+     * @return true/false in case of success/failure
+     */
+    virtual bool getIMUMeasurement(const std::string& imuName,
+                                   GenericContainer::Vector<double>& imuMeasurement,
+                                   double* receiveTimeInSeconds = nullptr) = 0;
+
+    /**
+     * Get linear accelerometer measurement in m/s^2
+     * @param[in] accName name of the linear accelerometer
+     * @param[out] accMeasurement linear accelerometer measurements of size 3
+     * @param[out] receiveTimeInSeconds time at which the measurement was received
+     * @return true/false in case of success/failure
+     */
+    virtual bool getLinearAccelerometerMeasurement(const std::string& accName,
+                                                   GenericContainer::Vector<double>& accMeasurement,
```

Comment:
Is it a good idea to use the `GenericContainer::Vector` here? In all the other parts of the code, we use Eigen vectors as intefrace of the class. I would avoid having mixing different kinds of types

Reply:
For instance here you can use an `Eigen::Ref<Eigen::Vector3d>` 
And the you can pass every different container (std vector iDynTree or yarp) through an `Eigen::Map`

Reply:
We had a discussion about this at https://github.com/dic-iit/bipedal-locomotion-framework/pull/87#issuecomment-682450524.

Reply:
The important point that was raised was,

> So, right now I would suggest using GenericContainer::Vector in case you need to deal with different vector implementation at the same time, especially in case you need to resize them (something that is not possible with an Eigen::map).

@S-Dafarra ,

> I would avoid having mixing different kinds of types

would you have an opinion on this?

Reply:
> @S-Dafarra ,
> 
> > I would avoid having mixing different kinds of types
> 
> would you have an opinion on this?

I would say that it depends on whether ``accMeasurement`` has a predefined size or not. If so, I agree on using Eigen types for consistency.

Reply:
There are a few quantities that have might have a variable size like images, joint states, etc. Otherwise, the rest are all fixed size.

Anyhow, these are going to remain fixed once the user configures the number of joints or the image dimensions.

Reply:
> There are a few quantities that have might have a variable size like images, joint states, etc. Otherwise, the rest are all fixed size.

The images I think are stored in matrices right? If yes in this case `GenericContainer::Vector` cannot be used. For the joint state back in time I simply used `iDynTree::VectorDynSize`, using Eigen the code may become
```cpp
bool SensorBridge::getJointPos(Eigen::Ref<Eigen::VectorXd> v);

....

main()
{
  SensorBridge sb;
  iDynTree::VectorDynSize q_iDynTree;
  yarp::Vector q_yarp;
  Eigen::VectorXd q_eigen;
  sb.getJointPos(iDynTree::toEigen(q_iDynTree));
  sb.getJointPos(yarp::toEigen(q_yarp));
  sb.getJointPos(yarp::toEigen(q_eigen));
}
```

Reply:
> > There are a few quantities that have might have a variable size like images, joint states, etc. Otherwise, the rest are all fixed size.
> 
> The images I think are stored in matrices right? If yes in this case `GenericContainer::Vector` cannot be used. For the joint state, back in time I simply used `iDynTree::VectorDynSize`, using Eigen the code may
> 
> ```c++
> bool SensorBridge::getJointPos(Eigen::Ref<Eigen::VectorXd> v);
> 
> ....
> 
> main()become
> {
>   SensorBridge sb;
>   iDynTree::VectorDynSize q_iDynTree;
>   yarp::Vector q_yarp;
>   Eigen::VectorXd q_eigen;
>   sb.getJointPos(iDynTree::toEigen(q_iDynTree));
>   sb.getJointPos(yarp::toEigen(q_yarp));
>   sb.getJointPos(q_eigen);
> }
> ```

By the way this holds only if we don't want to resize the vector inside the `getJointPos()` because in this case, terrible bugs may happen 
```cpp

Eigen::VectorXd q_eigen(100);
 
sb.getJointPos(q_eigen.segment<10>(20)); //<----- If getJointPos resize the vectors terrible things may happen here 
```

Reply:
Relevant changes made in https://github.com/dic-iit/bipedal-locomotion-framework/pull/87/commits/4a4b818f9bb3ad844993a7a6596c16844e7af942

Reply:
Thank you for handling this :heart:

Reply:
So, how is it handled for the elements with variable size?

Reply:
At the moment, we assume for joint states and images, that the size is decided at the configuration and remains fixed, and internal checks can be done at the implementation level. This means that the user must pass a resized argument to these getter methods. Correct me if I am wrong, @GiulioRomualdi!
I shall add a description in the documentation regarding this.

However, we need to understand how to handle this for the elements with variable size, for example, point clouds.

Reply:
Ok for the documentation. For point clouds, are they matrices or vector of points?

Reply:
They are usually handled as a struct asking for maintaining cloud width and height. Then they store a vector of XYZ points or XYZRGB points. 

I hope we will not be causing any overheads by doing multiple conversions by passing these kinds of images or point cloud measurements through this bridge.

Reply:
I think may open one or more issue to discuss on this. In particular, it is not clear:
- when to use ``Eigen`` vectors vs ``GenericContainer`` vs templates
- how to deal with matrices
- how to deal with particular input classes, like point clouds

Reply:
I have added documentation for the joint states and image methods and relevant configuration parameters in the `SensorBridgeOptions` if the implementation might want to use it. https://github.com/dic-iit/bipedal-locomotion-framework/pull/87/commits/979733e1357ffed832c1533a6c6c464adcdae6b8